core/src/config.jl

```diff
@@ -149,6 +144,13 @@ end
     results::Results = Results()
 end
 
+struct Config
+    toml::Toml
+    relative_dir::String
+end
```

Comment:
I like separating the TOML from other things. I don't like having to always type e.g. `config.toml.starttime`. Would it work for you if we forward `Base.getproperty` on `Config` to the `toml` field? To maintain the old API. To access the `relative_dir` field, which perhaps should just be `dir` or `directory`, we could add `Base.dirname(config::Config) = getfield(config, :relative_dir)`.

Reply:
> forward Base.getproperty on Config to the toml field

Nice idea. It adds a bit complexity, but I think it's worth it.

> relative_dir field, which perhaps should just be dir or directory

I think relative_dir is more descriptive

> we could add Base.dirname(config::Config) = getfield(config, :relative_dir)

The [docs](https://docs.julialang.org/en/v1/base/file/#Base.Filesystem.dirname) Base.dirname talk about paths.
Since config isn't a path, I don't think it is applicable here. I've created a function `relative_dir` instead.

> The Base.show method in this file probably also need updating.

It seems to work fine for me. It's what is called if I evaluate a `Config` instance in the REPL, right?

Reply:
Thanks! Not sure what you think, but the name `relative_dir` has always been a bit confusing to me. It is not necessarily relative, it is just a directory, it's other paths in the TOML that are relative to it.

Regarding `Base.dirname`, good that you checked the docstring for it. The line "Get the directory part of a path." really seems to describe the particular `::String` method though. If it is a true function docstring it should describe what is does ("get the directory"), not from where. And what it does seems to match fine. Wflow also has this method on their `Config`.

Indeed Base.show seems to work fine, I mistakenly assumed it would be broken by this PR.